microcosm_caching/decorators.py

```diff
@@ -101,12 +109,12 @@ def cache(*args, **kwargs) -> Schema:
                 return func(*args, **kwargs)
 
             try:
-                key = cache_key(cache_prefix, kwargs[identifier_key])
+                key = cache_key(cache_prefix, schema, args, kwargs)
                 cached_resource = retrieve_from_cache(key)
```

Comment:
from the time we check if the cached resource exists until the time we add_in_cache can have potential for race condition. this is fine for now and we can add some kind of semaphore in memcached if needed.

Reply:
Yeah. I have a set of 3s for now, which may not be enough if commits start taking longer.